examples/flight-animation.js

```diff
@@ -0,0 +1,135 @@
+// NOCOMPILE
+// this example uses arc.js for which we don't have an externs file.
+goog.require('ol.Attribution');
+goog.require('ol.Feature');
+goog.require('ol.Map');
+goog.require('ol.View');
+goog.require('ol.control');
+goog.require('ol.geom.LineString');
+goog.require('ol.layer.Tile');
+goog.require('ol.layer.Vector');
+goog.require('ol.proj');
+goog.require('ol.source.Stamen');
+goog.require('ol.source.Vector');
+goog.require('ol.style.Stroke');
+goog.require('ol.style.Style');
+
+var map = new ol.Map({
+  layers: [
+    new ol.layer.Tile({
+      source: new ol.source.Stamen({
+        layer: 'toner'
+      })
+    })
+  ],
+  controls: ol.control.defaults({
+    attributionOptions: /** @type {olx.control.AttributionOptions} */ ({
+      collapsible: false
+    })
+  }),
+  renderer: 'canvas',
+  target: 'map',
+  view: new ol.View({
+    center: [0, 0],
+    zoom: 2
+  })
+});
+
+var defaultStroke = new ol.style.Stroke({
+  color: '#EAE911',
+  width: 2
+});
+var defaultStyle = new ol.style.Style({
+  stroke: defaultStroke
+});
+
+var pointsPerMs = 0.1;
+var animateFlights = function(event) {
+  var vectorContext = event.vectorContext;
+  var frameState = event.frameState;
+  vectorContext.setFillStrokeStyle(null, defaultStroke);
+
+  var features = flightsSource.getFeatures();
+  for (var i = 0; i < features.length; i++) {
+    var feature = features[i];
+    if (!feature.get('finished')) {
+      // only draw the lines for which the animation has not
+      // finished yet
+      var coords = feature.getGeometry().getCoordinates();
+      var elapsedTime = frameState.time - feature.get('start');
+      var elapsedPoints = elapsedTime * pointsPerMs;
+
+      if (elapsedPoints >= coords.length) {
+        feature.set('finished', true);
+      }
+
+      var maxIndex = Math.min(elapsedPoints, coords.length);
+      var currentLine = new ol.geom.LineString(coords.slice(0, maxIndex));
+
+      // directly draw the line with the vector context
+      vectorContext.drawLineStringGeometry(currentLine, feature);
+    }
+  }
+  // tell OL3 to continue the postcompose animation
+  frameState.animate = true;
```

Comment:
Instead of having people mutate the frame state, I think we should encourage the use of `map.render()` (this is part of the documented API).